packages/tool-cache/src/tool-cache.ts

```diff
@@ -206,6 +207,29 @@ export async function extractTar(
   return dest
 }
 
+/**
+ * Extract a xar compatible archive
+ *
+ * @param file     path to the archive
+ * @param dest     destination directory. Optional.
+ * @param flags    flags for the xar. Optional.
+ * @returns        path to the destination directory
+ */
+export async function extractXar(
+  file: string,
+  dest?: string,
+  flags: string = '-x'
+): Promise<string> {
+  ok(IS_MAC, 'extractXar() not supported on current OS')
+  ok(file, 'parameter "file" is required')
+
+  dest = dest || (await _createExtractFolder(dest))
+  const xarPath: string = await io.which('xar', true)
+  await exec(`"${xarPath}"`, [flags, '-C', dest, '-f', file])
```

Comment:
considering adding the -v (verbose) flag if debug is set.

[Example](https://github.com/actions/toolkit/blob/master/packages/tool-cache/src/tool-cache.ts#L231)